realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -59,19 +61,46 @@ private RealmThreadPoolExecutor(int corePoolSize, int maxPoolSize) {
                 new ArrayBlockingQueue<Runnable>(QUEUE_SIZE));
     }
 
-    @Override
-    public Future<?> submit(Runnable task) {
+    /**
+     * Submits a runnable for executing a transaction.
+     *
+     * @param task the task to submit
+     * @return a future representing pending completion of the task
+     */
+    public Future<?> submitTransaction(Realm realm, Runnable task) {
```

Comment:
BTW, if they take the same param `submitTransaction`/`submitQuery` and do the exact same thing, no need to split into two functions ....